### ISSUE TITLE
alertType = .force issue with checkVersion type daily, weekly

### ISSUE THREAD

**Issue by kamrankhan07** [ArtSabintsev/Siren#237]
In method showAlert(), check date is stored immediately using method storeVersionCheckDate(), which is then used later to decide whether to check new version or not using method checkVersion(checkType: VersionCheckType).
Lets say user never uses the type .immediately and has code for weekly checks, and let say set the major version change to force user to update. So, after a week the code will go and check for new version and it happens to be a major version change, Siren will show Update alert to force user. But if user kills the app and opens it up again, it will not show force update message because the lastVersionCheckPerformedOnDate date would have been updated in showAlert() and the next check will be performed after a week. In my opinion that is not desirable. I propose that we only store the date if alert type is not force 

```
func showAlert() {
        
        if alertType != .force {
            storeVersionCheckDate()
        }
```

Please let me know, if there can be any side effects of this. I thought about it and I don't seem to think of any. I already have a pull request, if you agree with this change then I can include it in that pull request.

**Comment by ArtSabintsev** [comment]
Alright, so usually I'd recommend that if you use `.force` to use `.immediately`, so in your situation, which I know is something you control through a server-side config file (due to #236), I think you should change both values if you decide to force a user to update.

I do see the issue you raise though, however, I need to think about it a bit before. 

I think this is related to this ticket: https://github.com/ArtSabintsev/Siren/issues/190

It  may be time to finally work on this feature.

**Comment by kamrankhan07** [comment]
I already do like that for force update case, always check with immediately if controlled from server side. However, this bug is a general case without involving server side, if users don't user .immediately as checkVersion type.

**Comment by ArtSabintsev** [comment]
I've set aside time to do this in early November. I'll keep this thread updated.

**Comment by ArtSabintsev** [comment]
I've begun working on Swift 4.0.0 in #250 - all I have left is to implement this feature.

**Comment by ArtSabintsev** [comment]
Now a secondary issue that arises when implementing this feature is that we now will have layers of rules:

- Performing a version check next time means show an alert next time, irrespective of what Siren is configured to do.
- Skipping a version check now needs to override all settings until the subsequent new version is checked for.

This makes this problem a lot more complex to solve now. I have to think about how I want to approach this feature since it will involve a bigger rewrite with a lot of potential bugs.

**Comment by ArtSabintsev** [comment]
ok, I've solved this issue in #250, however, I still need to add documentation and a README. Feel free to subscribe to that PR for updates. 

Closing this issue as it's essentially fixed.
